src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts

```diff
@@ -234,11 +262,49 @@ export class GKEClusterSettingsComponent
       .subscribe((zones: string[]) => (this.zones = zones));
   }
 
+  private _getGKEDiskTypes(): void {
+    this._externalClusterService
+      .getGKEDiskTypes(this.controlValue(Controls.Zone))
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe((diskTypes: GCPDiskType[]) => {
+        diskTypes.map(diskType => {
+          if (diskType.name === this.DISK_TYPE_DEFAULT_VALUE) {
+            this.control(Controls.DiskTypes).setValue(diskType.name);
+          }
+        });
+        this.diskTypes = diskTypes;
+        this.diskTypeLabel = this.diskTypes?.length ? DiskTypeState.Ready : DiskTypeState.Empty;
+      });
+  }
+
+  private _getGKEMachineTypes(): void {
+    this._externalClusterService
+      .getGKEMachineTypes(this.controlValue(Controls.Zone))
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe((machineTypes: GCPMachineSize[]) => {
+        machineTypes.map(machineType => {
+          if (machineType.name === this.MACHINE_TYPE_DEFAULT_VALUE) {
+            this.control(Controls.MachineTypes).setValue(machineType.name);
+          }
+        });
+        this.machineTypes = machineTypes;
+        this.machineTypeLabel = this.machineTypes?.length ? MachineTypeState.Ready : MachineTypeState.Empty;
+      });
+  }
+
   private _getGKEDiskTypesForMachineDeployment(): void {
+    this.diskTypeLabel = DiskTypeState.Loading;
     this._externalMachineDeploymentService
       .getGKEDiskTypesForMachineDeployment(this.projectID, this.cluster.id)
-      .pipe(map((diskTypes: GCPDiskType[]) => diskTypes.map(type => type.name + ` (${type.description})`)))
-      .subscribe((diskTypes: string[]) => (this.diskTypes = diskTypes));
+      .subscribe((diskTypes: GCPDiskType[]) => {
+        diskTypes.map(diskType => {
```

Comment:
Check this comment: https://github.com/kubermatic/dashboard/pull/4917#discussion_r960396002